### PR TITLE
Add fmt and clippy

### PR DESCRIPTION
**Pull request by alvicsam**
Can I suggest using gitlab-ci instead of GHA?

Related to:
https://github.com/paritytech/cumulus/issues/614
https://github.com/paritytech/cumulus/pull/615

### CODE REVIEW

**Comment by chevdor** [comment]
What would be the benefit in this case ?

**Comment by alvicsam** [comment]
> What would be the benefit in this case ?

Almost all our projects use gitlab-ci and in future we want to have unified pipeline for all projects (at least the repeatable steps). So it's better to have one unified instrument for CI.

**Comment by alvicsam** [comment]
Found out that `paritytech/ci-linux` doesn't have not fmt, nor clippy >_<
But I'll fix it.

**Comment by chevdor** [comment]
For now, I am mainly working on fixing the issues related to those checks. 

Whether we do the checks in github or gitlab can be discussed. I would prioritize simplicity and speed, this is why I went for GHA since this is supposed to be a **quick** check and not a big mechanics. 

Ultimately, I don't mind but I find GHA more convenient for those, especially because we could have one Github Action that we use adn reused in most repos.

For big and heavy runs, I am with you on the choice of gitlab though.

NOTE: In your pipeline, you should run `fmt` first as it runs much faster and thus can fail much faster.

**Comment by alvicsam** [comment]
`fmt` took 2 minutes, so it's relatively quick :) And it's just 5 lines in `.giltab-ci.yml`

>NOTE: In your pipeline, you should run fmt first as it runs much faster and thus can fail much faster.

Currently they are in one stage, so they run simultaneously. 

p.s. still need to add clippy to ci image

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -67,6 +68,23 @@ variables:                         &default-vars
     paths:
       - ./artifacts/
 
+#### stage:                        lint
+
+fmt:
+  stage:                           lint
+  <<:                              *docker-env
```
Comment (rcny) [inline .gitlab-ci.yml:75]: `&docker-env` refers to `linux-docker` runners - we use them for heavyweight jobs (e.g. compilation stuff). These runners have limited capacity. Please resort to `&kubernetes-env` (`kubernetes-parity-build`) instead.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -67,6 +68,23 @@ variables:                         &default-vars
...
+  script:
+    - cargo +nightly fmt --all -- --check
+
+clippy:
+  stage:                           lint
+  <<:                              *docker-env
```
Comment (rcny) [inline .gitlab-ci.yml:81]: Same as above.

**Review by rcny — CHANGES_REQUESTED** [review]

**Comment by rcny** [comment]
`&kubernetes-env` lacks `image: "${CI_IMAGE}"` directive.

**Review by rcny — APPROVED** [review]

**Comment by bkchr** [comment]
Already done. Ty anyway. 

I know that this runs stuff on gitlab,but I don't see any advantage in this anyway, so I'm going to close this pr.

**Pull request closed without merging**